Doc/c-api/init.rst

```diff
@@ -2152,3 +2157,41 @@ be used in new code.
 .. c:function:: void PyThread_delete_key_value(int key)
 .. c:function:: void PyThread_ReInitTLS()
 
+Synchronization Primitives
+==========================
+
+The C-API provides a basic mutual exclusion lock.
+
+.. c:type:: PyMutex
+
+   A mutual exclusion lock.  The :c:type:`!PyMutex` should be initialized to
+   zero to represent the unlocked state.  For example::
+
+      PyMutex mutex = {0};
+
+   Instances of :c:type:`!PyMutex` should not be copied or moved.  Both the
+   contents and address of a :c:type:`!PyMutex` are meaningful, and it must
+   remain at a fixed, writable location in memory.
+
+   .. note::
+
+      A :c:type:`!PyMutex` currently occupies one byte, but the size should be
+      considered unstable.  The size may change in future Python releases
+      without a deprecation period.
+
+   .. versionadded:: 3.13
```

Comment:
We are past 3.13 beta1. Did you get an exception to get this feature in Python 3.13? If no, you should target Python 3.14.

Reply:
@Yhg1s is it acceptable to expose `PyMutex` as a public C API in 3.13? There are docs and refactoring to make it public, but no new functionality.

(For context, the only other free-threading related API I'd like to see public in 3.13 is the [critical section API](https://github.com/python/cpython/issues/119344), but there isn't a decision yet on that from the WG)

Reply:
Yes, considering the use-case, the extent of the discussion around it, and the fact that this is exposing an existing (albeit new in 3.13) API, this is fine to go in before beta 3.